src/openrct2/rct2/RCT2.h

```diff
@@ -786,6 +789,263 @@ struct RCT2RideRatingCalculationData
+
+    // SC6[8]
+    uint16_t last_guests_in_park;
+    uint8_t pad_01357BCA[3];
+    uint8_t handyman_colour;
+    uint8_t mechanic_colour;
+    uint8_t security_colour;
+
+    // Ignored in scenario
+    uint32_t researched_scenery_items[RCT2_MAX_RESEARCHED_SCENERY_ITEM_QUADS];
+
+    // SC6[9]
+    uint16_t park_rating;
+
+    // Ignored in scenario
+    uint8_t park_rating_history[32];
+    uint8_t guests_in_park_history[32];
+
+    // SC6[10]
+    uint8_t active_research_types;
+    uint8_t research_progress_stage;
+    uint32_t last_researched_item_subject;
+    uint8_t pad_01357CF8[1000];
+    uint32_t next_research_item;
+    uint16_t research_progress;
+    uint8_t next_research_category;
+    uint8_t next_research_expected_day;
+    uint8_t next_research_expected_month;
+    uint8_t guest_initial_happiness;
+    uint16_t park_size;
+    uint16_t guest_generation_probability;
+    uint16_t total_ride_value_for_money;
+    money32 maximum_loan;
+    money16 guest_initial_cash;
+    uint8_t guest_initial_hunger;
+    uint8_t guest_initial_thirst;
+    uint8_t objective_type;
+    uint8_t objective_year;
+    uint8_t pad_013580FA[2];
+    money32 objective_currency;
+    uint16_t objective_guests;
+    uint8_t campaign_weeks_left[20];
+    uint8_t campaign_ride_index[22];
+
+    // Ignored in scenario
+    money32 balance_history[RCT12_FINANCE_GRAPH_SIZE];
+
+    // SC6[11]
+    money32 current_expenditure;
+    money32 current_profit;
+    money32 weekly_profit_average_dividend;
+    uint16_t weekly_profit_average_divisor;
+    uint8_t pad_0135833A[2];
+
+    // Ignored in scenario
+    money32 weekly_profit_history[RCT12_FINANCE_GRAPH_SIZE];
+
+    // SC6[12]
+    money32 park_value;
+
+    // Ignored in scenario
+    money32 park_value_history[RCT12_FINANCE_GRAPH_SIZE];
+
+    // SC6[13]
+    money32 completed_company_value;
+    uint32_t total_admissions;
+    money32 income_from_admissions;
+    money32 company_value;
+    uint8_t peep_warning_throttle[16];
+    rct12_award awards[RCT12_MAX_AWARDS];
+    money16 land_price;
+    money16 construction_rights_price;
+    uint16_t word_01358774;
+    uint8_t pad_01358776[2];
+    uint32_t cd_key;
+    uint8_t pad_0135877C[64];
+    uint32_t game_version_number;
+    money32 completed_company_value_record;
+    uint32_t loan_hash;
+    uint16_t ride_count;
+    uint8_t pad_013587CA[6];
+    money32 historical_profit;
+    uint8_t pad_013587D4[4];
+    char scenario_completed_name[32];
+    money32 cash;
+    uint8_t pad_013587FC[50];
+    uint16_t park_rating_casualty_penalty;
+    uint16_t map_size_units;
+    uint16_t map_size_minus_2;
+    uint16_t map_size;
+    uint16_t map_max_xy;
+    uint32_t same_price_throughout;
+    uint16_t suggested_max_guests;
+    uint16_t park_rating_warning_days;
+    uint8_t last_entrance_style;
+    uint8_t rct1_water_colour;
+    uint8_t pad_01358842[2];
+    RCT12ResearchItem research_items[RCT2_MAX_RESEARCH_ITEMS];
```

Comment:
This variable name suggests it's the max for both RCT1 and RCT2. Should the constant not reflect that?

Reply:
The structure is the same for both rct1 and 2 but the limits are different.